util/random_test.go

```diff
@@ -0,0 +1,32 @@
+package util
+
+import (
+	"testing"
+	"time"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetRandomTime(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct{
+		lowerBound time.Duration
+		upperBound time.Duration
+	}{
+		{ 1 * time.Second, 10 * time.Second },
```

Comment:
Might be worth testing a bunch of different ranges here, such as:

* (0, 0)
* (-1, -3)
* (1, 1.000001)
* (1, 10)

It may also be worth running a loop for each test case and testing it multiple times, as there are random numbers involved, so it's possible a test will pass as a pure fluke simply because the random number didn't catch a corner case.